src/TextField/EnhancedTextarea.js

```diff
@@ -75,6 +75,12 @@ class EnhancedTextarea extends Component {
     }
   }
 
+  componentDidUpdate(prevProps) {
```

Comment:
Yes, the actual problem is, that `syncHeightWithShadow` is using `this.props.rowsMax`, which will contain the value before the property was changed. Thus the height also calculated the height for the old value.

This is the "culprit":
```
    if (this.props.rowsMax >= this.props.rows) {
      newHeight = Math.min(this.props.rowsMax * rowsHeight, newHeight);
    }
```

Another solution to this can be to call `syncHeightWithShadow` with the correct `props` and using them instead of `this.props` -- `syncHeightWithShadow(newValue, event, props)`.

I think its a slightly better solution, since `syncHeightWithShadow` won't be called twice.